datafusion/core/tests/sql/set_variable.rs

```diff
@@ -260,37 +260,238 @@ async fn set_u64_variable_bad_value() {
+        .unwrap();
+
+    // casting UTF-8 to TimestampTZ isn't supported yet, add Timestamp as the middle layer for now
+    let result =
+        plan_and_collect(&ctx, "SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ")
+            .await
+            .unwrap();
+    let expected = vec![
+        "+-----------------------------------------------------+",
+        "| Utf8(\"2000-01-01T00:00:00\")                         |",
+        "+-----------------------------------------------------+",
+        "| 2000-01-01T00:00:00 (Unknown Time Zone '+08:00:00') |",
+        "+-----------------------------------------------------+",
+    ];
+    assert_batches_eq!(expected, &result);
+
+    plan_and_collect(&ctx, "SET TIME ZONE = '08:00'")
+        .await
+        .unwrap();
+
+    // casting UTF-8 to TimestampTZ isn't supported yet, add Timestamp as the middle layer for now
+    let result =
+        plan_and_collect(&ctx, "SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ")
+            .await
+            .unwrap();
+    let expected = vec![
+        "+-------------------------------------------------+",
+        "| Utf8(\"2000-01-01T00:00:00\")                     |",
+        "+-------------------------------------------------+",
+        "| 2000-01-01T00:00:00 (Unknown Time Zone '08:00') |",
+        "+-------------------------------------------------+",
+    ];
+    assert_batches_eq!(expected, &result);
+
+    plan_and_collect(&ctx, "SET TIME ZONE = '08'")
+        .await
+        .unwrap();
+
+    // casting UTF-8 to TimestampTZ isn't supported yet, add Timestamp as the middle layer for now
+    let result =
+        plan_and_collect(&ctx, "SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ")
+            .await
+            .unwrap();
+    let expected = vec![
+        "+----------------------------------------------+",
+        "| Utf8(\"2000-01-01T00:00:00\")                  |",
+        "+----------------------------------------------+",
+        "| 2000-01-01T00:00:00 (Unknown Time Zone '08') |",
+        "+----------------------------------------------+",
+    ];
+    assert_batches_eq!(expected, &result);
+
+    // we dont support named time zone yet
+    plan_and_collect(&ctx, "SET TIME ZONE = 'Asia/Taipei'")
+        .await
+        .unwrap();
+
+    // casting UTF-8 to TimestampTZ isn't supported yet, add Timestamp as the middle layer for now
+    let result =
+        plan_and_collect(&ctx, "SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ")
+            .await
+            .unwrap();
+    let expected = vec![
+        "+-------------------------------------------------------+",
+        "| Utf8(\"2000-01-01T00:00:00\")                           |",
+        "+-------------------------------------------------------+",
+        "| 2000-01-01T00:00:00 (Unknown Time Zone 'Asia/Taipei') |",
+        "+-------------------------------------------------------+",
+    ];
+    assert_batches_eq!(expected, &result);
```

Comment:
@avantgardnerio 
based on the discussion here https://github.com/apache/arrow-datafusion/issues/4106#issuecomment-1303856668

named timezone like `America/Denver` or `Asia/Taipei` won't be supported in this pr